src/modules/home/index.tsx

```diff
@@ -1,16 +1,29 @@
 import { lazyRoute } from '#/utils/router.util';
 import { type RouteObject } from 'react-router-dom';
-import { WishlistRoute } from './routes/wishlist.route';
+import { AuthLayout } from '#/routes/layout-protected.route';
 
 const { HomeRoute } = lazyRoute(() => import('./routes/home.route'), 'HomeRoute');
+const { WishlistRoute } = lazyRoute(() => import('./routes/wishlist.route'), 'WishlistRoute');
 
 export const HomeRoutes: RouteObject[] = [
   {
     path: '/',
-    element: <HomeRoute />,
+    element: <AuthLayout protectedRoute={false} />,
+    children: [
+      {
+        path: '',
+        element: <HomeRoute />,
+      },
+    ],
   },
   {
     path: '/wishlist',
-    element: <WishlistRoute />,
+    element: <AuthLayout protectedRoute={true} />,
```

Comment:
If you're already protecting private routes in `src/routes/router.tsx` why is doing it again here necessary? Doesn't this make the same auth check twice? auth Or did I understand the code incorrectly?

Reply:
there are three scenarios when it comes to protecting routes:
1. the route is public and doesn't need user info
2. the route is public but needs user info (to display the name for example)
3. the route is private and requires authentication (`/wishlist` route for example)

the `protectedRoute` prop helps differentiate between second and third cases. About `src/router/router.tsx`, this is actually why I was asking your opinion on this. The thing is that `private.route.ts` and `public.route.ts` take in routes based on modules (home module routes and auth module routes). But the problem is that each module can have both public and private routes (root is public, wishlist is private in home module), even though the module is public as a whole. this makes it necessary to go to module level and specifically wrap individual routes with `AuthLayout` if they are private (or need user info).

Reply:
We can go with what @bek-shoyatbek suggested.